src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbol.cs

```diff
@@ -225,13 +225,16 @@ public sealed override RefKind RefKind
         /// <summary>
         /// The declared scope. From source, this is from the <c>scope</c> keyword only.
         /// </summary>
-        internal DeclarationScope DeclaredScope => _scope;
+        internal DeclarationScope? DeclaredScope => _scope;
 
         internal abstract override DeclarationScope EffectiveScope { get; }
 
         protected DeclarationScope CalculateEffectiveScopeIgnoringAttributes()
         {
-            var declaredScope = this.DeclaredScope;
+            // DeclaredScope is only null in LambdaParameterSymbol where EffectiveScope can be provided rather than computed
+            Debug.Assert(this.DeclaredScope is not null);
+            var declaredScope = this.DeclaredScope.Value;
```

Comment:
>Value [](http://example.com/codeflow?start=51&length=5)

```ValueOrDefault```? Or is it your intent to crash on ```null``` value here? If so, perhaps the assert is redundant.

Reply:
I'd rather leave as-is. I thought this was nullable-enabled code.